servicetalk-concurrent-api/docs/modules/ROOT/pages/blocking-implementation.adoc

```diff
@@ -20,77 +21,186 @@ Everything inside ServiceTalk is somehow connected to one of the three asynchron
+event loop thread, which is owned by a Netty `IOExecutor`. Assuming that the default ServiceTalk offloading has been
+disabled, it will only offload at the `publishOn()` operator and transition execution from the event loop thread to an
+`Executor` thread. Once the publish signal is offloaded the event loop thread will be available again for executing
+other I/O tasks while the response is asynchronously processed on the `Executor` thread.
+
+=== `publishOn()` Example
+
+Using the `publishOn(executor)` operator allows processing of signals related to the source content on a different
+thread than is generating (or receiving) the content. It is the most common form of offloading used as it relates to
+the data path, specifically the `Subscription` methods.
+
+[source, java]
+----
+Collection<Integer> result = Publisher.range(1, 10) <2> <5>
+        .map(element -> element)  // non-offloaded NO-OP
+        .publishOn(publishExecutor)  <4>
+        .map(element -> element)  // offloaded NO-OP
+        .toFuture()  <3>
+        .get();  <1> <6>
+----
 
-ServiceTalk concurrency APIs defines which thread will be used by any asynchronous source for Part 2.
+<1> `toFuture().get` begins by calling `subscribe(Subscriber)`. Executing on the application thread, execution flows up
+the operator chain towards the source; `map` -> `publishOn` -> `map` -> `Publisher.range(1, 10)`.
 
-== Executor and asynchronous sources
+<2> Still executing on the application thread, `Range` will call `Subscriber.onSubscribe(Subscription)` on the
+`Subscriber`. This flows back down the operator chain, `Range` -> `map` -> `publishOn` -> `map` -> `toFuture`.
 
-ServiceTalk enforces that each asynchronous source **MUST** be associated with an `Executor` which is used to interact
-with its `Subscriber`. The below diagram illustrates the interaction between an asynchronous source, its `Subscriber`,
-its operators, and the `Executor`.
+<3> When `subscribe()` is complete, `toFuture()` will call `Subscription.request(Long.MAX_VALUE)`. This will cause
+`Range` to publish synchronously via `onNext(element)` nine items, the integers "`1`" through "`9`".
 
-image::executor-thread-selection.png[Executor inference]
+<4> Each `onNext` flows down the operator chain, `Range` -> `map` -> `publishOn` (offloads on to `publishExecutor`
+thread) -> `map` -> `toFuture` where the element is collected for the result. For each offloaded item a thread of
+`publishExecutor` will be used for executing the second `map` operator and final collect operation.
 
-The above interaction provides all operators and the source an opportunity to piggy-back on the same `Executor` for all
-interactions with the `Subscriber`. Any operator in a chain has two options while accepting a `Subscriber`:
+<5> The application thread will continue to publish all the elements from the source and will finally send the
```

Comment:
`The application thread will continue to pub` -> use a consistent approach as above to describe control flow:

`onComplete()` flows down the operator chain, `Range` -> `map` -> `publishOn` (offloads on to `publishExecutor`
 thread) -> `map` -> `toFuture` ...